samplesheets/tests/test_views_api_taskflow.py

```diff
@@ -144,6 +240,645 @@ def test_post_created(self):
+            response = self.client.put(
+                reverse(
+                    'samplesheets:api_irods_request_update',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                ),
+                update_data,
+            )
+
+        self.assertEqual(response.status_code, 400)
+        obj.refresh_from_db()
+        self.assertEqual(obj.description, 'bla')
+        self.assertEqual(obj.path, self.path)
+
+
+class TestIrodsRequestDeleteAPIView(TestIrodsRequestAPIViewBase):
+    """Tests for IrodsRequestDeleteAPIView"""
+
+    def test_delete(self):
+        """Test deleting a request"""
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:api_irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+
+        self.assertEqual(IrodsDataRequest.objects.count(), 1)
+        obj = IrodsDataRequest.objects.first()
+
+        with self.login(self.user_contrib):
+            response = self.client.delete(
+                reverse(
+                    'samplesheets:api_irods_request_delete',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                )
+            )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+
+    def test_delete_one_of_multiple(self):
+        """Test deleting one of multiple requests"""
+        path2 = os.path.join(self.assay_path, TEST_FILE_NAME + '_2')
+        path2_md5 = os.path.join(self.assay_path, TEST_FILE_NAME + '_2.md5')
+        self.irods.data_objects.create(path2)
+        self.irods.data_objects.create(path2_md5)
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:api_irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.post_data['path'] = path2
+            self.client.post(
+                reverse(
+                    'samplesheets:api_irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 2)
+            obj = IrodsDataRequest.objects.first()
+            response = self.client.delete(
+                reverse(
+                    'samplesheets:api_irods_request_delete',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                )
+            )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(IrodsDataRequest.objects.count(), 1)
+        self.assert_alert_count(CREATE_ALERT, self.user, 1)
```

Comment:
We are still looking for `CREATE_ALERT` alerts here for a delete method. Probably not that relevant here? If we create alerts about request deletion (I'm not sure if we do?) then those should be tested instead.

Reply:
Just missed this pair :(